source/brailleInput.py

```diff
@@ -451,16 +451,17 @@ def handleCaretMove(self, obj):
 		self.flushBuffer()
 
 	def handlePostConfigProfileSwitch(self):
-		if config.conf["braille"]["inputTable"] == "auto":
-			config.conf["braille"]["inputTable"] = brailleTables.getDefaultTableForCurLang(
-				brailleTables.TableType.INPUT,
-			)
 		# #6140: Migrate to new table names as smoothly as possible.
 		tableName = config.conf["braille"]["inputTable"]
 		newTableName = brailleTables.RENAMED_TABLES.get(tableName)
 		if newTableName:
 			tableName = config.conf["braille"]["inputTable"] = newTableName
-		table = config.conf["braille"]["inputTable"]
+		if config.conf["braille"]["inputTable"] == "auto":
+			table = brailleTables.getDefaultTableForCurLang(
```

Comment:
it looks like the whole assignment to `table` is redundant. Could you have another look at this code? I think you should also be able to avoid the else block below altogether, since tableName already has the table.
```suggestion
			tableName = brailleTables.getDefaultTableForCurLang(
```